### ISSUE TITLE
Provide strongly-typed `DSL.table` factory methods

### ISSUE THREAD

**Issue by DavidGregory084** [jOOQ/jOOQ#16867]
### Use case

Due to some [TimescaleDB](https://docs.timescale.com/use-timescale/latest/continuous-aggregates/time/#time-and-continuous-aggregates) limitations I have found it necessary to create several very similar views which differ only in the timezone used to calculate time buckets for my data.

I'm able to use jOOQ code generation to query these very similar views, but this currently requires lots of branching expressions to determine the correct view to query based upon the summary period and the time zone relative to which the data must be summarised.

### Possible solution

Provide new overloads of the `DSL.table` method which allow a user to construct a `Table<T>` given only its name and record type:

```java
<T extends Record> Table<T> table(String name, Class<T> recordType)
<T extends Record> Table<T> table(Name name, Class<T> recordType)
<T extends Record> Table<T> table(SQL name, Class<T> recordType)
```

These are analogous to the existing `DSL.field` overloads which accept a `Class<T>` or `DataType<T>` and return a `Field<T>`, and serve a similar purpose in enabling users to query tables whose record types are known statically without requiring the use of code generation.

### Possible workarounds

Use jOOQ code generation and write a lot of branching statements to pick the appropriate generated table definition.

### jOOQ Version

jOOQ Open Source 3.19.8

### Database product and version

PostgreSQL 15.6 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit

### Java Version

openjdk 17.0.10 2024-01-16

### JDBC / R2DBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.6.0

**Comment by lukaseder** [comment]
Thanks for your suggestion.

We already have this on our roadmap:

- https://github.com/jOOQ/jOOQ/issues/7444

It will solve this problem more thoroughly than your suggestion. Adding more overloads to `DSL.table()` like this won't scale in terms of API design.

Closing this as a duplicate of #7444

**Comment by lukaseder** [comment]
Note, see also:

- https://github.com/jOOQ/jOOQ/issues/9594

This has a higher chance of being implemented soon, compared to #7444

**Comment by DavidGregory084** [comment]
> > We already have this on our roadmap:
> 
>     * [Create a public wrapper around the TableImpl and related meta data API #7444](https://github.com/jOOQ/jOOQ/issues/7444)

Thanks for this, `TableImpl` definitely works for my use case so it would certainly be great to have a public interface for it!
